backend/entities/office_hours/oh_event_entity.py

```diff
@@ -0,0 +1,59 @@
+"""Definition of SQLAlchemy table-backed object mapping entity for Office Hour Sections."""
+
+from datetime import datetime
+from sqlalchemy import DateTime, ForeignKey, Integer, String
+from sqlalchemy.orm import Mapped, mapped_column, relationship
+
+from backend.models.office_hours.oh_type import OfficeHoursType
+
+
+from ..entity_base import EntityBase
+from typing import Self
+from sqlalchemy import Enum as SQLAlchemyEnum
+
+__authors__ = ["Madelyn Andrews", "Sadie Amato", "Bailey DeSouza", "Meghan Sun"]
+__copyright__ = "Copyright 2024"
+__license__ = "MIT"
+
+
+class OfficeHoursEventEntity(EntityBase):
+    """Serves as the database model schema defining the shape of the `OfficeHoursEvent` table"""
+
+    # Name for the events table in the PostgreSQL database
+    __tablename__ = "office_hours__event"
+
+    # Unique id for OfficeHoursEvent
+    id: Mapped[int] = mapped_column(Integer, primary_key=True, autoincrement=True)
+    # Type of event
+    type: Mapped[OfficeHoursType] = mapped_column(
+        SQLAlchemyEnum(OfficeHoursType), nullable=False
+    )
+    # Name of event
+    title: Mapped[str] = mapped_column(String, nullable=False)
+    # Description of event
+    description: Mapped[str] = mapped_column(String, nullable=True)
+    # Description of the location; allows for instructors to write note about attending office hours
+    location_description: Mapped[str] = mapped_column(String, nullable=True)
+    # Date of the event
+    date: Mapped[datetime] = mapped_column(DateTime, nullable=False)
```

Comment:
Thinking more on this field, I think we should use the `date` class instead of the datetime class since we have a start and end time as separate fields. See info on the classes [here](https://docs.python.org/3/library/datetime.html)
The line could look like this: `date: Mapped[date] = mapped_column(Date, nullable=False)`